### PR TITLE
bqml pipeline components notebook official blog

### PR DESCRIPTION
**Pull request by inardini**
If you are opening a PR for `Official Notebooks` under the [notebooks/official](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/official) folder, follow this mandatory checklist:
- [x] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [x] Follow the style and grammar rules outlined in the above notebook template.
- [x] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [x] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/docs/contributing.md#code-quality-checks).
- [ ] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [x] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/docs/CODEOWNERS) file under `# Official Notebooks` section, pointing to the author or the author's team.
- [x] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

### CODE REVIEW

File: notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb [hunk 1]
```diff
@@ -0,0 +1,1460 @@
+{
```
Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: <p>Add a link to open notebook in Vertex AI Workbench </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/403/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: <p>branding:</p><p>BQML -&gt; BigQuery ML</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/403/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: <p>is it necessary to pin versions for?: google-cloud-aiplatform==1.10.0 google_cloud_pipeline_components==1.0.0&nbsp;</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/403/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: <p>REGION = "[your-region]"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/403/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    BUCKET_URI = "gs://mlops-bqml-text-analyisis"  # @param {type:"string"}
</pre><p>BUCKET_URI = gs://[your-bucket-name]"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/403/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: <p>could do this as a single mkdir command</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/403/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='31'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: <p>Add the steps here, like:</p><p> </p><p>1) Get dataset from UCI archive.</p><p>2) Untar the dataset</p><p>3) Copy the dataset to a Cloud Storage location.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/403/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='32'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: <p>Imports for components should be moved to inside the pipeline definition</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/403/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='36'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: <p>describe purpose</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/403/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='43'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: <p>describe purpose</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/403/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='48'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: <p>I see you write out the SQL queries to files, but never use the files?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/403/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='53'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: <p>describe purpose</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/403/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='54'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: <p>describe purpose</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/403/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='56'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: <p>describe purpose</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/403/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='61'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #5.    def pipeline(
</pre><p>the bq queries should be passed in as arguments -- not hardwired.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/403/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='67'/>

Comment (inardini) [reply notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: <p>I was thinking about reproducibility Andrew. But I can unpin if you want</p>

Comment (inardini) [reply notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: <p>done</p>

Comment (inardini) [reply notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: <p>I was thinking about the best practices. Probably I would have all .sql scripts for each component of  BQ pipeline. In a way that I can load them when I build it</p>

Comment (inardini) [reply notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: <p>Done</p>

Comment (inardini) [reply notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: done

Comment (inardini) [reply notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: done

Comment (inardini) [reply notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: done

Comment (inardini) [reply notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: done

[the remark this answers is not in the document]
Comment (inardini) [reply notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: done

Comment (inardini) [reply notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: done

Comment (inardini) [reply notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: done

[the remark this answers is not in the document]
Comment (inardini) [reply notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: done

Comment (inardini) [reply notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: done

[the remark this answers is not in the document]
Comment (inardini) [reply notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: done

Comment (inardini) [reply notebooks/official/pipelines/google_cloud_pipeline_components_bqml_text.ipynb:1]: done

**Pull request closed without merging**